File: src/domain/repositories/SupportTicketRepository.ts
import type { SupportTicket } from '@/domain/entities/SupportTicket'

export interface ResolveSupportTicketInput {
  ticketId: string
  response: string
  resolvedById: string
  resolvedByName: string
}

export interface SupportTicketRepository {
  listAll(): Promise<SupportTicket[]>
  resolve(input: ResolveSupportTicketInput): Promise<SupportTicket>
}

File: consorcio/lib/domain/repositories/support_ticket_repository.dart
import '../entities/support_ticket.dart';

abstract class SupportTicketRepository {
  Future<SupportTicket> create({
    required String kind,
    required String message,
    required String createdById,
    required String createdByName,
  });

  Future<List<SupportTicket>> listMine(String userId);

  Future<List<SupportTicket>> listAll();

  Future<SupportTicket> resolve({
    required String ticketId,
    required String response,
    required String resolvedById,
    required String resolvedByName,
  });
}
